chef/cookbooks/crowbar/recipes/prepare-upgrade-scripts.rb

```diff
@@ -188,16 +188,17 @@
   action :create
 end
 
+neutron_server = search(:node, "run_list_map:neutron-server").first
 template "/etc/neutron/lbaas-connection.conf" do
   source "lbaas-connection.conf"
   mode "0640"
   owner "root"
   group "root"
   action :create
   variables(
-    lazy do
-      { sql_connection: node[:neutron][:db][:sql_connection] }
-    end
+    lazy {
+      { sql_connection: neutron_server[:neutron][:db][:sql_connection] }
+    }
   )
   only_if { roles.include? "neutron-network" }
```

Comment:
i thought about that. As it is right now, the file will be created without the sql_connection information if that neutron-server is missing. Otherwise the file won't be created at all.

The first case helped me identify that the script was actually running but with wrong arguments so it was fast to find the cause. The latter could be a bit misleading.

In either case, both will fail miserably because we don't handle this situation nicely. I'd prefer to leave as it is now, which is mirroring the nova-controller code section.